### PR TITLE
Fix config mapping for runtime and frontend targets for RelWithDebugInfo config

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Fix for RelWithDebugInfo config in samples to bind it with openvino::runtime Release instead of Debug on Windows.
 
### Tickets:
 - CVS-109331

### CODE REVIEW

File: samples/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -206,8 +206,18 @@ macro(ie_add_sample)
     find_package(OpenVINO REQUIRED COMPONENTS Runtime)
     if(c_sample)
         set(ov_link_libraries openvino::runtime::c)
+        get_target_property(ORIGINAL_NAME openvino::runtime::c ALIASED_TARGET)
+        if (ORIGINAL_NAME MATCHES "ORIGINAL_NAME-NOTFOUND")
+            set (ORIGINAL_NAME "openvino::runtime::c")
+        endif()
+        set_property(TARGET ${ORIGINAL_NAME} PROPERTY "MAP_IMPORTED_CONFIG_RELWITHDEBINFO" "RELEASE")
```
[the remark this answers is not in the document]
Comment (olpipi) [reply samples/cpp/CMakeLists.txt:213]: I'll try it

[the remark this answers is not in the document]
Comment (olpipi) [reply samples/cpp/CMakeLists.txt]: done

File: cmake/templates/OpenVINOConfig.cmake.in [hunk 2]
```diff
@@ -283,6 +283,18 @@ if(NOT TARGET openvino)
...
 
+    foreach(target openvino::runtime openvino::runtime::c
+               openvino::frontend::onnx openvino::frontend::paddle openvino::frontend::tensorflow
+               openvino::frontend::pytorch openvino::frontend::tensorflow_lite)
+        if(TARGET ${target})
+            get_target_property(ORIGINAL_NAME ${target} ALIASED_TARGET)
+            if (ORIGINAL_NAME MATCHES "ORIGINAL_NAME-NOTFOUND")
+                set (ORIGINAL_NAME ${target})
+            endif()
+            set_property(TARGET ${ORIGINAL_NAME} PROPERTY "MAP_IMPORTED_CONFIG_RELWITHDEBINFO" "RELEASE")
```
Comment (ilya-lavrenov) [inline cmake/templates/OpenVINOConfig.cmake.in:294]: what if RelWithDebInfo is installed? 
We need to check whether RELWITHDEBINFO is among of [IMPORTED_CONFIGURATIONS](https://cmake.org/cmake/help/latest/prop_tgt/IMPORTED_CONFIGURATIONS.html)

File: cmake/templates/OpenVINOConfig.cmake.in [hunk 3]
```diff
@@ -283,6 +283,18 @@ if(NOT TARGET openvino)
     set(_ov_as_external_package ON)
     include("${CMAKE_CURRENT_LIST_DIR}/OpenVINOTargets.cmake")
 
+    foreach(target openvino::runtime openvino::runtime::c
+               openvino::frontend::onnx openvino::frontend::paddle openvino::frontend::tensorflow
+               openvino::frontend::pytorch openvino::frontend::tensorflow_lite)
```
Comment (ilya-lavrenov) [inline cmake/templates/OpenVINOConfig.cmake.in:288]: we also have a list of all targets here:
https://github.com/openvinotoolkit/openvino/blob/master/cmake/templates/OpenVINOConfig.cmake.in#L329-L331
let's declare a list of targets in a single place and reuse in both places.
Later not to forget to `unset` this variable.

Comment (olpipi) [reply cmake/templates/OpenVINOConfig.cmake.in:288]: The list you mentioned does not include pytorch and tensorflow_lite frontends. Do I need to add them here?

Comment (ilya-lavrenov) [reply cmake/templates/OpenVINOConfig.cmake.in:288]: yes, it's a mistake that those libs are missed

Comment (olpipi) [reply cmake/templates/OpenVINOConfig.cmake.in]: done

File: cmake/templates/OpenVINOConfig.cmake.in (hunk 2 above)
Comment (olpipi) [reply cmake/templates/OpenVINOConfig.cmake.in]: done

**Review by vurusovs — APPROVED** [review]

**Pull request merged**